includes/MediaWiki/Request/EditRequestParser.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\Request;
+
+use MediaWiki\Extension\WikibaseReconcileEdit\InputToEntity\FullWikibaseItemInput;
+use MediaWiki\Extension\WikibaseReconcileEdit\InputToEntity\MinimalItemInput;
+use MediaWiki\Rest\LocalizedHttpException;
+use MediaWiki\Rest\RequestInterface;
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\DataModel\Services\Lookup\PropertyDataTypeLookup;
+use Wikimedia\Message\MessageValue;
+
+/**
+ * @license GPL-2.0-or-later
+ */
+class EditRequestParser {
+
+	public const VERSION_KEY = "wikibasereconcileedit-version";
+
+	private const SUPPORTED_ENTITY_VERSIONS = [ '0.0.1/full', '0.0.1/minimal' ];
+	private const SUPPORTED_RECONCILE_VERSIONS = [ '0.0.1' ];
+
+	/** @var PropertyDataTypeLookup */
+	private $propertyDataTypeLookup;
+
+	/** @var FullWikibaseItemInput */
+	private $fullWikibaseItemInput;
+
+	/** @var MinimalItemInput */
+	private $minimalItemInput;
+
+	public function __construct(
+		PropertyDataTypeLookup $propertyDataTypeLookup,
+		FullWikibaseItemInput $fullWikibaseItemInput,
+		MinimalItemInput $minimalItemInput
+	) {
+		$this->propertyDataTypeLookup = $propertyDataTypeLookup;
+		$this->fullWikibaseItemInput = $fullWikibaseItemInput;
+		$this->minimalItemInput = $minimalItemInput;
+	}
+
+	public function parseRequestInterface( RequestInterface $request ): EditRequest {
```

Comment:
could be named just parse? Feels redundant having the name of the thing to be parsed in the method name.

Reply:
Well, my assumption is that we might parse something else later, for batch mode (where the API would get an array of arrays from the request parameters, and then pass those inner arrays into the parser).

Reply:
Ok, sounds good